lib/enigma.rb

```diff
@@ -37,4 +37,35 @@ def shifts(key, date)
     end
     shifts
   end
+
+  def to_indexes(string)
+    alphabet = Array("a".."z")<< " "
+    string_array = string.downcase.chars
+    string_array.map do |char|
+      alphabet.find_index(char)
+    end
+  end
+
+  def indexes_to_string(string, shifts)
+    alphabet = Array("a".."z")<< " "
+    shifts.map do |number|
+      alphabet.rotate(number)[0]
+    end.join
+  end
+
+  def cipher(string, shifts)
+    indexes_plus_shifts = []
+    to_indexes(string).each_with_index do |number, index|
+      indexes_plus_shifts << number + shifts[index%4]
+    end
+    indexes_to_string(string, indexes_plus_shifts)
+  end
```

Comment:
`cipher` takes an in both a string to be ciphered and an array of shift values with which to cipher the string. 
First it will take the string and find the index position of each character in the string with `to_index`.
Once we have the numbers array from `to_index` we can then iterate over those numbers adding each one of our shift values to each number in the array based on the index positions of each array and shovel the results into an accumulator.
Finally we take our string and the accumulator and pass them in as arguments for `indexes_to_string`.